src/app-layer-ftp.c

```diff
@@ -455,18 +511,23 @@ static int FTPParseRequest(Flow *f, void *ftp_state,
                     memcpy(data->file_name, state->current_line + 5, state->current_line_len - 5);
                     data->cmd = state->command;
                     data->flow_id = FlowGetId(f);
-                    int ret = AppLayerExpectationCreate(f, direction, 0,
-                                                         state->dyn_port,
-                                                         ALPROTO_FTPDATA, data);
+                    int ret = AppLayerExpectationCreate(f,
+                                            state->active ? STREAM_TOSERVER : direction,
```

Comment:
Should state->active be enough to say if it is to_server or to_client ? Why do we need direction here ?

Reply:
My understanding is we're setting up an expectation for the data channel in which case the direction is flipped for active mode.

Is that correct?

Reply:
Yes I was thinking you could do `state->active ? STREAM_TOSERVER : STREAM_TOCLIENT` or something alike.

Reply:
I'm not sure if that would be more clear.